src/mine_loop.rs

```diff
@@ -673,6 +672,7 @@ pub(crate) async fn mine(
     mut from_main: mpsc::Receiver<MainToMiner>,
     to_main: mpsc::Sender<MinerToMain>,
     mut global_state_lock: GlobalStateLock,
+    perform_initial_sleep: bool,
```

Comment:
Ugh, boolean arguments… 🚩

Is this only set to `false` in tests? If so, how about `#[cfg(not(test))]`-ing the offending line? If it's used somewhere else, how about turning this into an argument `initial_sleep: Duration`?

Reply:
It's presently only used for tests.  I think cleanest for now may be the initial_sleep: Duration.  will change.

aside:
though it feels like polishing a turd.

The entire concept of the initial sleep is a mostly useless hack imho that will hopefully soon go away.   It is a hard-coded, arbitrary 60 second sleep and the comment says it is for waiting for the node to sync up with peers, but that could take from seconds to days.  

and beyond that the entire mine() fn needs a redesign / re-write.

iirc, the mining state-machine PR I made dispenses with the initial sleep.  But I'm not super happy with that PR due to complexity... it likely just makes things harder to understand.

Reply:
I decided to go with #[cfg(not(test))].   It seems simplest option for now.

Reply:
addressed in 1b9a52926cb7755fc853048ec76c5d066a32d47d

Reply:
> it feels like polishing a turd

😅

> The entire concept of the initial sleep will hopefully soon go away.

🤞